### PR TITLE
msdfgen: add 1.11 + package_type + bump deps

### PR DESCRIPTION
**Pull request by SpaceIm**
also use cache_variables in order to avoid patching of upstream CMakeLists (since 1.10)

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
```
[100%] Linking CXX executable msdfgen
/opt/conan/binutils/bin/ld: warning: libbrotlienc.so.1, needed by /home/conan/w/prod-v2/bsr/40266/bddbb/p/freetf759ac655ea20/p/lib/libfreetype.so.6, not found (try using -rpath or -rpath-link)
/opt/conan/binutils/bin/ld: warning: libbrotlidec.so.1, needed by /home/conan/w/prod-v2/bsr/40266/bddbb/p/freetf759ac655ea20/p/lib/libfreetype.so.6, not found (try using -rpath or -rpath-link)
/opt/conan/binutils/bin/ld: warning: libbrotlicommon.so.1, needed by /home/conan/w/prod-v2/bsr/40266/bddbb/p/freetf759ac655ea20/p/lib/libfreetype.so.6, not found (try using -rpath or -rpath-link)
/opt/conan/binutils/bin/ld: /home/conan/w/prod-v2/bsr/40266/bddbb/p/libpn702506f471cbf/p/lib/libpng16.so.16: undefined reference to `inflateValidate@ZLIB_1.2.9'
/opt/conan/binutils/bin/ld: /home/conan/w/prod-v2/bsr/40266/bddbb/p/freetf759ac655ea20/p/lib/libfreetype.so.6: undefined reference to `BrotliDecoderDecompress'
```

Another occurrence of https://github.com/conan-io/conan/issues/13560

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by valgur — APPROVED** [review]

File: recipes/msdfgen/all/conanfile.py [hunk 1]
```diff
@@ -49,11 +49,11 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("freetype/2.12.1")
+        self.requires("freetype/2.13.2")
         if  Version(self.version) < "1.10":
             self.requires("lodepng/cci.20200615")
         else:
-            self.requires("libpng/1.6.39")
+            self.requires("libpng/1.6.40")
```
Comment (mayeut) [inline recipes/msdfgen/all/conanfile.py:56]:
```suggestion
            self.requires("libpng/[>=1.6 <2]")
```
version range is now allowed for libpng

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**